### PR TITLE
Update `StackOutputs` fields

### PR DESCRIPTION
**Pull request by Overcastan**
This PR updates the fields of `StackOutputs` struct. 
It changes them from `Vec<u64>` to `Vec<Felt>` to make them consistent with the stack values.

### CODE REVIEW

**Review by hackaugusto — APPROVED** [review]
LGTM

File: miden/src/examples/mod.rs [hunk 1]
```diff
@@ -170,7 +170,7 @@ where
     let program_info = ProgramInfo::new(program.hash(), kernel);
 
     if fail {
-        outputs.stack_mut()[0] += 1;
+        outputs.stack_mut()[0] += Felt::new(1);
```
Comment (bobbinth) [inline miden/src/examples/mod.rs:173]: We can use just `ONE` here.

File: test-utils/src/lib.rs [hunk 2]
```diff
@@ -252,7 +252,7 @@ impl Test {
 
         let program_info = ProgramInfo::from(program);
         if test_fail {
-            stack_outputs.stack_mut()[0] += 1;
+            stack_outputs.stack_mut()[0] += Felt::new(1);
```
Comment (bobbinth) [inline test-utils/src/lib.rs:255]: Same comment as above.

File: core/src/stack/outputs.rs [hunk 3]
```diff
@@ -77,15 +61,38 @@ impl StackOutputs {
...
-        let stack = stack.iter().map(|&v| v.as_int()).collect::<Vec<_>>();
-        let overflow_addrs = overflow_addrs.iter().map(|&v| v.as_int()).collect::<Vec<_>>();
+    /// Attempts to create [StackOutputs] struct from the provided stack elements and overflow
+    /// addresses represented as vectors of `u64` values.
+    ///
+    /// # Errors
+    /// Returns an error if:
+    /// - Any of the provided stack elements are invalid field elements.
+    /// - Any of the provided overflow addresses are invalid field elements.
+    pub fn try_from_values(stack: Vec<u64>, overflow_addrs: Vec<u64>) -> Result<Self, OutputError> {
```
Comment (bobbinth) [inline core/src/stack/outputs.rs:82]: nit: I would rename this to `try_from_ints`.

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you! I left a couple of small nits inline - after these, we can merge.

**Comment by bobbinth** [comment]
Oh - and also, let's update the changelog.

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,7 @@
 
 #### VM Internals
 - Removed unused `find_lone_leaf()` function from the Advice Provider (#1262).
+- Changed fields type of the `StackOutputs` struct from `Vec<u64>` to `Vec<Felt>` (#1268).
```
Comment (bobbinth) [inline CHANGELOG.md:7]: This is a breaking change (i.e., it changes the public interface). We should mark it as such.

Comment (bobbinth) [reply CHANGELOG.md:7]: Yep - that's fine.

**Pull request merged**